client/src/app/users/user.service.spec.ts

```diff
@@ -59,135 +60,237 @@ describe('User service: ', () => {
+
+          // This gets the arguments for the first (and in this case only) call to the `mockMethod`.
+          const [url, options] = mockedMethod.calls.argsFor(0);
+          // Gets the `HttpParams` from the options part of the call.
+          // `options.param` can return any of a broad number of types;
+          // it is in fact an instance of `HttpParams`, and I need to use
+          // that fact, so I'm casting it (the `as HttpParams` bit).
+          const calledHttpParams: HttpParams = (options.params) as HttpParams;
+          expect(mockedMethod)
+            .withContext('one call')
+            .toHaveBeenCalledTimes(1);
+          expect(url)
+            .withContext('talks to the correct endpoint')
+            .toEqual(userService.userUrl);
+          expect(calledHttpParams.keys().length)
+            .withContext('should have 3 params')
+            .toEqual(3);
+          expect(calledHttpParams.get('role'))
+            .withContext('role of editor')
+            .toEqual('editor');
+          expect(calledHttpParams.get('company'))
+            .withContext('company being IBM')
+            .toEqual('IBM');
+          expect(calledHttpParams.get('age'))
+            .withContext('age being 37')
+            .toEqual('37');
+        });
+    });
   });
 
-  it('getUserById() calls api/users/id', () => {
-    const targetUser: User = testUsers[1];
-    const targetId: string = targetUser._id;
-    userService.getUserById(targetId).subscribe(
-      user => expect(user).toBe(targetUser)
-    );
-
-    const expectedUrl = `${userService.userUrl}/${targetId}`;
-    const req = httpTestingController.expectOne(expectedUrl);
-    expect(req.request.method).toEqual('GET');
-    req.flush(targetUser);
+  describe('When getUserByID() is given an ID', () => {
+    it('calls api/users/id with the correct ID', waitForAsync(() => {
+      // We're just picking a User "at random" from our little
+      // set of Users up at the top.
+      const targetUser: User = testUsers[1];
+      const targetId: string = targetUser._id;
+
+      // Mock the `httpClient.get()` method so that instead of making an HTTP request
+      // it just returns our test data
+      const mockedMethod = spyOn(httpClient, 'get').and.returnValue(of(targetUser));
+
+      // Call `userService.getUser()` and confirm that the correct call has
+      // been made with the correct arguments.
+      //
+      // We have to `subscribe()` to the `Observable` returned by `getUserById()`.
+      // The `user` argument in the function below is the thing of type User returned by
+      // the call to `getUserById()`.
+      userService.getUserById(targetId).subscribe((user: User) => {
+        // The `User` returned by `getUserById()` should be targetUser.
+        // This `expect` doesn't do a _whole_ lot.
+        // This really just confirms that `getUserById()`
+        // doesn't in some way modify the user it
+        // gets back from the server.
```

Comment:
```
        // This `expect` doesn't do a _whole_ lot.
        // This really just confirms that `getUserById()`
        // doesn't in some way modify the user it
        // gets back from the server.
```

This is the kind of comment that we should keep and repeat in all the places where we have these "unnecessary" tests.